cranelift/fuzzgen/src/lib.rs

```diff
@@ -48,18 +50,21 @@ impl fmt::Debug for TestCase {
 
         // Print only non default flags
         let default_flags = Flags::new(settings::builder());
-        for (default, flag) in default_flags.iter().zip(self.flags.iter()) {
+        for (default, flag) in default_flags.iter().zip(self.isa.flags().iter()) {
             assert_eq!(default.name, flag.name);
 
             if default.value_string() != flag.value_string() {
                 writeln!(f, "set {}={}", flag.name, flag.value_string())?;
             }
         }
 
-        writeln!(f, "target aarch64")?;
-        writeln!(f, "target s390x")?;
-        writeln!(f, "target riscv64")?;
-        writeln!(f, "target x86_64\n")?;
+        match self.isa.triple().architecture {
+            Architecture::X86_64 => writeln!(f, "target aarch64")?,
+            Architecture::Aarch64 { .. } => writeln!(f, "target aarch64")?,
+            Architecture::S390x => writeln!(f, "target s390x")?,
+            Architecture::Riscv64 { .. } => writeln!(f, "target riscv64")?,
+            _ => unreachable!(),
+        }
```

Comment:
@afonso360 is this what you had in mind?

Reply:
Yeah that looks good!

The reason for that is mostly that when we generate a function it is only guaranteed to compile for a particular architecture (due to unimplemented instructions), and having all of them there is slightly misleading.